tests/object_cache_test.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+
+from clvm.object_cache import ObjectCache, treehash, serialized_length
+
+from clvm_tools.binutils import assemble
+
+
+class ObjectCacheTest(unittest.TestCase):
+    def check(self, obj_text, expected_hash, expected_length):
+        obj = assemble(obj_text)
```

Comment:
I think circular dependency of `clvm_tools` should be avoided.
How about removing `assemble` and replacing with `to_sexp_type` here?

```python
class ObjectCacheTest(unittest.TestCase):
    def check(self, obj, expected_hash, expected_length):
        th = ObjectCache(treehash)
        self.assertEqual(th.get(obj).hex(), expected_hash)
        sl = ObjectCache(serialized_length)
        self.assertEqual(sl.get(obj), expected_length)

    def test_various(self):
        self.check(
            # "0x00"
            to_sexp_f(b'\x00'),
            "47dc540c94ceb704a23875c11273e16bb0b8a87aed84de911f2133568115f254",
            1,
        )
        self.check(
            # "0"
            to_sexp_f(None),
            "4bf5122f344554c53bde2ebb8cd2b7e3d1600ad631c385a5d7cce23c7785459a", 1
        )
        self.check(
            # "foo"
            to_sexp_f("foo"),
            "0080b50a51ecd0ccfaaa4d49dba866fe58724f18445d30202bafb03e21eef6cb", 4
        )
        self.check(
            # "(foo . bar)",
            to_sexp_f(("foo", "bar")),
            "c518e45ae6a7b4146017b7a1d81639051b132f1f5572ce3088a3898a9ed1280b",
            9,
        )
        self.check(
            # "(this is a longer test of a deeper tree)"
            to_sexp_f([b"this", b"is", b"\x02", b"longer", b"test", b"of", b"\x02", b"deeper", b"tree"]),
            "0a072d7d860d77d8e290ced0fdb29a271198ca3db54d701c45d831e3aae6422c",
            47,
        )
```

Reply:
The tests already depend upon `clvm_tools` as they use `brun` internally. I agree though, you're right that I should use `to_sexp_f`.